.github/workflows/tests.yml

```diff
@@ -25,10 +25,12 @@ jobs:
         with:
           version: 3.x
           repo-token: ${{ secrets.GITHUB_TOKEN }}
-      - name: Run Tests
+      - name: Lint code
```

Comment:
This isn't clear to me, why does the workflow have to have a linting step? I thought when the workflow runs it should expect the code to already be linted. 

To me it would make more sense for the `Test code` step to also do `task setup` and for this `Lint code` step to not be added.

Reply:
`task lint` verifies that code has been linted - which is lacking from this CI

`task test` only checks a fraction of the code in this repo whereas `task lint` reviews all the go code.
Also, testing doesn't catch a long list of things that could be problematic but don't actively cause problems.

For example
```go
// golangci-lint points out that "Error return value of `json.Unmarshal` is not checked"
json.Unmarshal([]byte(data), &result)

// Which would result in us updating the above to this below
if err := json.Unmarshal([]byte(data), &result); err != nil {
	panic(err)
}
```
Both would pass in our tests, but one is much easier to debug.